### ISSUE TITLE
Use a different default data directory than Ethereum

### ISSUE THREAD

**Issue by ethernomad** [ethereumproject/go-ethereum#100]
Any time I run the Ethereum Classic Geth without specifying a data directory separate from Ethereum, it wipes Ethereum's data directory. This is extremely irritating as then I have to download the entire Ethereum blockchain again from scratch.

**Comment by splix** [comment]
@ethernomad thank you for reporting this. This is know and really annoying issue, we're going to work on this soon. 

For now the general workaround is to specify chain dir with `--datadir <path>` option

**Comment by outragedhuman** [comment]
@splix I've submitted #109 for this. As noted in the request, I see this has a size/L label, so now I'm a bit concerned I missed something big. If so, please let me know and apologies in advance.

**Comment by splix** [comment]
Hi @outragedhuman. Thank you for the patch. 

The reason why it have size/L is because it should also check non-classic dir for existing data, and if data is here and belongs to classic chain it should rename existing dir to new one. And do that only once, of course, when target dir not-exists yet.

We need that because downloading whole chain takes days, users don't want to resync just because we use different dir now. And also that dir contains private keys, people will be really upset if they will lost them with previous installation.

Maybe should be size/M, but anyway

**Comment by splix** [comment]
Sorry I didn't add my concerns before you've started

**Comment by outragedhuman** [comment]
@splix With the additional explanation the label seems more appropriate :)

No need to apologize, I had a feeling there was a reason! I am trying to learn Go, so I will go back to my fork and see if I can work that in as well.

**Comment by outragedhuman** [comment]
@splix After additional review, I certainly agree with the size/L label and it looks like this would be better handled by someone more familiar with Go.

I'll definitely be digging around for other stuff to fix though :)

One note that might help someone else fix it:

https://github.com/ethereumproject/go-ethereum/blob/65883388f6564419e27f8a6f1f4c45da6c095e75/cmd/geth/chaincmd.go#L140

upgradeDb() may be useful for backing up the old default dir and moving it to the new Classic dir.

**Comment by realcodywburns** [comment]
From a UX perspective, having to add a ton of flags or write a bash script is very inconvenient and leads to problems such as this. A default stateful config file would be very helpful. This probably needs its own issue.

**Comment by splix** [comment]
in context of #131 we need a root dir for chains, and a subdir per chain
```
~/ethereum-classic
-  mainnet (data, keys, etc, all here)
   testnet
   my-custom-net 
   .....
```

**Comment by whilei** [comment]
Closing via #189.
